### PR TITLE
openvc: 4.x modernize more for conan v2

### PR DESCRIPTION
**Pull request by prince-chrismc**
Specify library name and version:  **lib/1.0**

Building this locally I ran into a version conflict

```
Graph root
    cli
Requirements
    imath/3.1.6#7f6b123c01e44d29f5170f2141083e44 - Downloaded (conancenter)
    jasper/4.0.0#478d874438e50163be99725c8603281a - Downloaded (conancenter)
    jbig/20160605#2d29fa02aacd76902e0d2cbbc24631ef - Downloaded (conancenter)
    libdeflate/1.17#71f492ef993b45d9b961b36fb85e801f - Downloaded (conancenter)
    libjpeg/9e#68269859e4325ddc3f995d1fd3fc9187 - Downloaded (conancenter)
    libpng/1.6.39#deca555b0890e584fc895ba756e83c06 - Downloaded (conancenter)
    libtiff/4.4.0#e6d8942022c89a22dfbe1a9b77289c7e - Downloaded (conancenter)
    opencv/4.5.5#107c7183c36c63faeb1512659ccd6c70 - Cache
    openexr/3.1.5#88ba2322e127ebe5635fcbfdff4928d2 - Downloaded (conancenter)
    xz_utils/5.4.0#186e94237b66f785e4ea095f2f621a56 - Downloaded (conancenter)
    zlib/1.2.13#13c96f538b52e1600c40b88994de240f - Downloaded (conancenter)
    zstd/1.5.4#15ac0bc440f89247276d78ea4fdf8b14 - Downloaded (conancenter)
ERROR: Version conflict: libtiff/4.4.0->libwebp/1.3.0, opencv/4.5.5->libwebp/1.2.4.
```

---

- [ ] I've read the [contributing guidelines](https://github.com/conan-io/conan-center-index/blob/master/CONTRIBUTING.md).
- [ ] I've used a [recent](https://github.com/conan-io/conan/releases/latest) Conan client version close to the [currently deployed](https://github.com/conan-io/conan-center-index/blob/master/.c3i/config_v1.yml#L6).
- [ ] I've tried at least one configuration locally with the [conan-center hook](https://github.com/conan-io/hooks.git) activated.

### CODE REVIEW

**Comment by nicosmd** [comment]
Hey @prince-chrismc, 
I currently have the same issue with the build in #16587. Can you tell me how I can resolve the kind of issue in conan center? Thanks!

**Review by nicosmd — APPROVED** [review]

**Comment by prince-chrismc** [comment]
#16797 should be ready now :)

**Comment by prince-chrismc** [comment]
```
  Setting up libxdamage-dev:amd64 (1:1.1.4-2) ...
   Setting up libxml2-utils (2.9.3+dfsg1-1ubuntu0.7) ...
   Setting up libgtk2.0-dev (2.24.30-1ubuntu1.16.04.2) ...
   Processing triggers for libc-bin (2.23-0ubuntu11.3) ...
   
   
   ======== Installing packages ========
   ERROR: Missing binary: quirc/1.1:2923dea4049294bf613203302e8065f31f47de77
   
   quirc/1.1: WARN: Can't find a 'quirc/1.1' package binary '2923dea4049294bf613203302e8065f31f47de77' for the configuration:
   [settings]
   arch=x86_64
   build_type=Release
   compiler=gcc
   compiler.version=11
   os=Linux
   [options]
   fPIC=True
   max_regions=254
   shared=False
   
   ERROR: Missing prebuilt package for 'quirc/1.1'
   Check the available packages using 'conan list quirc/1.1:* -r=remote'
   or try to build locally from sources using the '--build=quirc/1.1' argument
   
````

**Comment by prince-chrismc** [comment]
🤞  for https://github.com/conan-io/conan-center-index/pull/16805

**Comment by SpaceIm** [comment]
I have this huge & old PR https://github.com/conan-io/conan-center-index/pull/15193 fixing opencv 4.x recipe for conan v2 among other things, and you'll see that another fix is needed: `self.options["gtk"].version` must be replaced by `self.dependencies["gtk"].ref.version`

**Comment by prince-chrismc** [comment]
Thanks for the tip, that PR is way to big and need lots of time... my goal for the next week is to push as many v2 packages out. I'd love to get the bare minimum done for this to work in V2 for people to start using it 🚀

**Review by MartinDelille — APPROVED** [review]

**Comment by prince-chrismc** [comment]
working on https://github.com/conan-io/conan-center-index/pull/16894

**Comment by prince-chrismc** [comment]
Thats done lets try again

**Comment by SpaceIm** [comment]
test package of v2 pipeline, Linux / all shared:

```
/opt/conan/binutils/bin/ld: warning: libprotobuf.so.3.17.1.0, needed by /home/conan/w/prod-v2/BuildSingleReference/p/openc5269726356b93/p/lib/libopencv_dnn.so.4.5, not found (try using -rpath or -rpath-link)
```

Another occurrence for https://github.com/conan-io/conan/issues/13560

**Review by RubenRBS — CHANGES_REQUESTED** [review]
```
# TODO: use self.dependencies once https://github.com/conan-io/conan/issues/12728 fixed
                ffmpeg_component_version = self.deps_cpp_info["ffmpeg"].components[component].version
``` 

`ffmpeg` now properly sets `component_version` for each component. Can you update it? It has been reported as an issue in https://github.com/conan-io/conan/issues/13755

**Review by RubenRBS — APPROVED** [review]
Sure :)

**Review by MartinDelille — DISMISSED** [review]

File: recipes/opencv/4.x/conanfile.py [hunk 1]
```diff
@@ -219,7 +220,8 @@ def requirements(self):
         if self.options.get_safe("with_gtk"):
             self.requires("gtk/system")
         if self.options.dnn:
-            self.requires(f"protobuf/{self._protobuf_version}")
+            # Symbols are exposed https://github.com/conan-io/conan-center-index/pull/16678#issuecomment-1507811867
+            self.requires(f"protobuf/{self._protobuf_version}", transitive_libs=True)
```
Comment (SpaceIm) [inline recipes/opencv/4.x/conanfile.py:224]: it's not true, it's likely a bug of conan client.

**Comment by prince-chrismc** [comment]
racing condition https://github.com/conan-io/conan-center-index/commit/1e0d2484a5530db5bb63153a72aed90454fdc883 🤢

**Comment by SSE4** [comment]
> racing condition [1e0d248](https://github.com/conan-io/conan-center-index/commit/1e0d2484a5530db5bb63153a72aed90454fdc883) nauseated_face

can you explain?

**Comment by prince-chrismc** [comment]
> > racing condition [1e0d248](https://github.com/conan-io/conan-center-index/commit/1e0d2484a5530db5bb63153a72aed90454fdc883) nauseated_face
> 
> can you explain?

the recipe and most of the packages got uploaded and indexed but there was a few still indexing and being uploaded so some configs had missing packages, the revision changed in the middle of a build

**Comment by SpaceIm** [comment]
Is it possible to wait for merge of https://github.com/conan-io/conan-center-index/pull/17065 where ffmpeg builds fine in v2 pipeline now, and don't disable ffmpeg in this PR please?

**Comment by jcar87** [comment]
> Is it possible to wait for merge of #17065 where ffmpeg builds fine in v2 pipeline now, and don't disable ffmpeg in this PR please?

done!

File: recipes/opencv/4.x/conanfile.py (hunk 1 above)
Comment (SpaceIm) [reply recipes/opencv/4.x/conanfile.py:223]: Is it possible to change this comment? It's wrong. It's a workaround and it should be removed when fixed in conan client.

**Review by czoido — APPROVED** [review]

**Review by SSE4 — APPROVED** [review]

**Review by franramirez688 — APPROVED** [review]

**Review by davidsanfal — APPROVED** [review]

**Review by uilianries — APPROVED** [review]
Lgtm

**Review by RubenRBS — APPROVED** [review]

**Pull request closed without merging**